light/beacon/beacon_chain.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package beacon
+
+import (
+	"encoding/binary"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	// beacon header fields
+	BhiSlot          = 8
+	BhiProposerIndex = 9
+	BhiParentRoot    = 10
+	BhiStateRoot     = 11
+	BhiBodyRoot      = 12
+
+	// beacon state fields
+	BsiGenesisTime       = 32
+	BsiGenesisValidators = 33
+	BsiForkVersion       = 141
+	BsiLatestHeader      = 36
+	BsiBlockRoots        = 37
+	BsiStateRoots        = 38
+	BsiHistoricRoots     = 39
+	BsiFinalBlock        = 105
+	BsiSyncCommittee     = 54
+	BsiNextSyncCommittee = 55
+	BsiExecHead          = 908
+)
+
+var BsiFinalExecHash = ChildIndex(ChildIndex(BsiFinalBlock, BhiStateRoot), BsiExecHead)
+
+// Header defined a beacon header and supports JSON encoding according to the standard beacon API format
+type Header struct {
```

Comment:
We can define a bunch of functions of Header, like `epoch`, `period` etc, it will make code looks like more readable.

Reply:
I defined Epoch and SyncPeriod (also added a separate json struct to the api package because the `common.Decimal` <-> `uint64` conversions everywhere were really ugly)